docs/architecture/adr-074-timeout-params.md

```diff
@@ -0,0 +1,231 @@
+in Tendermint would mean entirely removing these parameters from any configuration
+that could be altered by either an application or a node operator. Instead,
+Tendermint would ship with a set of timeouts and all applications using Tendermint
+would use this exact same set of values.
+
+While Tendermint nodes often run with similar bandwidth and on similar cloud-hosted
+machines, there are enough points of variability to make configuring
+consensus timeouts meaningful. Namely, Tendermint network topologies are likely to be
+very different from chain to chain. Additionally, applications may vary greatly in 
+how long the `Commit` phase may take. Applications that perform more work during `Commit`
+require a longer `TimeoutCommit` to allow the application to complete its work
+and be prepared for the next height.
+
+* From https://github.com/tendermint/spec/issues/359 it looks like `TimeoutCommit` really only exists for the purpose of allowing
+precommits to be collected after the +2/3 has initially been seen. Not much more value to it than that.
+
+* allowing long timeout commit discussed here: https://github.com/tendermint/tendermint/issues/5911#issuecomment-973560381
+
+## Decision
+
+None
+
+## Detailed Design
+
+### New Consensus Parameters
+
+A new `TimeoutParams` `message` will be added to the [params.proto file][consensus-params-proto].
+This message will have the following form:
+
+```proto
+message TimeoutParams {
+ google.protobuf.Duration propose = 1;
+ google.protobuf.Duration propose_delta = 2;
+ google.protobuf.Duration vote = 3;
+ google.protobuf.Duration vote_delta = 4;
+ google.protobuf.Duration commit = 5;
+ bool commit_timeout_bypass_enabled = 6;
+}
+```
+
+This new message will be added as a field into the [`ConsensusParams`
+message][consensus-params-proto]. The same default values that are [currently
+set for these parameters][current-timeout-defaults] in the local configuration
+file will be used as the defaults for these new consensus parameters in the
+[consensus parameter defaults][default-consensus-params].
+
+The new consensus parameters will be subject to the same
+[validity rules][time-param-validation] as the current configuration values,
+namely, each value must be non-negative.
+
+### Migration
+
+Migration to using these new parameters will occur across two releases.
+During the v0.36 release, Tendermint will add these as consensus parameters.
+The old parameters config-local parameters will continue to exist as well in the
+v0.36 release. If the parameters are set as consensus parameters, the config-local
+parameters will be ignored. If the consensus parameters are not set, Tendermint
+will fall back to using the values contained in the `config.toml` file.
+
+During the v0.36 release cycle, if non-default values are set in the `config.toml`,
+the node will log a warning alerting operators that the parameters will soon be removed
+in the upcoming release.
```

Comment:
But more importantly, will these non-default values be used?

Reply:
They will only be used if the chain does not set them as consensus parameters.

Reply:
But what do you mean by not setting the consensus params. Aren't they set to some reasonable default upon initialization? How would they be unset?

Reply:
So basically it would be as follows:
If the chain has not yet set them in EndBlock, use what is in `config.toml`.
Once the chain sets them in `EndBlock`, use those values and ignore `config.toml`.

Reply:
Would this work with state syncing? Will we need an extra flag to tell us if it has been set or not? I think it would be better if we transition to using consensus params directly when the chain starts/restarts i.e. use the genesis values or the defaults in the case of new params in an upgraded chain (I know you've written an entire RFC on this so perhaps I should check there first)

Reply:
> Would this work with state syncing? Will we need an extra flag to tell us if it has been set or not?

We would not need this flag, no. We would just always treat 'value as absent' as 'value uses default'. This way historical blocks would always look to be using the defaults. Once the node that is syncing hits a height where the application changed the parameters via `EndBlock` it would then start to use the updated values. So far this is fine because all param changes we are proposing are timing related and validation of historical blocks does not consider timing at all.